src/main/java/org/rulez/demokracia/pdengine/BeatTable.java

```diff
@@ -0,0 +1,69 @@
+package org.rulez.demokracia.pdengine;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.rulez.demokracia.pdengine.dataobjects.Pair;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.types.MapMatrix;
+import org.rulez.demokracia.types.Matrix;
+
+public class BeatTable extends MapMatrix<Choice, Pair> implements Matrix<Choice, Pair>{
+	private static final long serialVersionUID = 1L;
+
+	public enum Direction {
+		DIRECTION_FORWARD, DIRECTION_BACKWARD
+	}
+
+	public BeatTable() {
+		this(new ArrayList<Choice>());
+	}
+
+	public BeatTable(final Collection<Choice> keyCollection) {
+		super(keyCollection);
+	}
+
+	public int beatInformation(final Choice choice1, final Choice choice2, final Direction direction) {
```

Comment:
Factor out everything which is not getter/setter or constructor to a treat (interface with default methods.